### DISCUSSION TITLE
How to make root-fs larger when RuntimeSize is configured?

### DISCUSSION THREAD

**Discussion by fwilhe2** [systemd/mkosi#2209] [Q&A] [answered]
Hi, this might be my missing experience when working with systemd-repart but I could not figure out how to resolve this.

What I want is pretty simple: I'm building a dev vm using mkosi and I need a larger root-fs for building software inside the VM.

I fund the `RuntimeSize` parameter and setting that properly enlarges the disk image when booting the VM as expected, but the root fs is not enlarged, so I get a ESP and a root partition which only take up a small fraction of the disk. I could manually resize the partition but I think this should not be needed.

My question is: How can I configure mkosi so that given a `RuntimeSize` of 30Gig, my root partition takes up all remaining space? I'm fine when this is happening on first boot, but it should be automated.

I've tried configuring [repart](https://github.com/systemd/mkosi/blob/main/mkosi/resources/mkosi.md?plain=1#L2021) but that did not bring the expected result.

What's the most simple way to do this?

**Comment by daandemeyer** [comment, marked as answer]
Put the following in `mkosi.extra/usr/lib/repart.d/10-root.conf`:

```
[Partition]
Type=root
```

That should be sufficient (you might need a somewhat recent systemd as I remember a recent bugfix for this).

Although for dev vms using mkosi I would skip the `disk` output completely and just use the `directory` output instead which will use virtiofs so there will be no need to select a size in the first place.

**Reply by fwilhe2** [reply]
Hi @DaanDeMeyer 

thanks for the quick answer. I'm not quite sure what happened now as df and fdisk report a different size.

```
root@localhost:~# df -h
Filesystem      Size  Used Avail Use% Mounted on
/dev/sda2       2.3G  1.6G  691M  70% /
devtmpfs        4.0M     0  4.0M   0% /dev
tmpfs           3.9G   84K  3.9G   1% /dev/shm
efivarfs         56K   11K   41K  22% /sys/firmware/efi/efivars
tmpfs           1.6G  8.6M  1.6G   1% /run
tmpfs           5.0M     0  5.0M   0% /run/lock
tmpfs           1.0M   24K 1000K   3% /run/credentials/@system
/dev/sda1       511M  144M  368M  29% /boot
tmpfs           1.0M  4.0K 1020K   1% /run/credentials/systemd-tmpfiles-setup.service
tmpfs           1.0M  8.0K 1016K   1% /run/credentials/getty@tty1.service
tmpfs           1.0M  8.0K 1016K   1% /run/credentials/serial-getty@ttyS0.service
tmpfs           792M  4.0K  792M   1% /run/user/0
root@localhost:~# fdisk -l /dev/sda
Disk /dev/sda: 30 GiB, 32212254720 bytes, 62914560 sectors
Disk model: QEMU HARDDISK   
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: gpt

Device       Start      End  Sectors  Size Type
/dev/sda1     2048  1050623  1048576  512M EFI System
/dev/sda2  1050624 62914519 61863896 29.5G Linux root (x86-64)
```

Regarding the directory output: I have not considered this. I think for what I'm trying to do I need to run the system in qemu and if I understand that right the directory output is made for nspawn, right?

**Reply by daandemeyer** [reply]
@fwilhe2 The discrepancy is probably due to your distribution's systemd missing https://github.com/systemd/systemd/pull/30030

The directory output can now be used with `qemu` as well if you have `virtiofsd` installed on the host. Please try it out, it's much faster than the `disk` output.

**Reply by fwilhe2** [reply]
Thanks, I'll try that out.

Btw, my host system is Fedora 39, and I'm building a VM for Debian Testing. I'm using mkosi from the Fedora repo which is currently at version 19, running with `systemd 254 (254.7-1.fc39)`.

**Reply by Cornelicorn** [reply]
The commit from that MR is only in 255 yet, sadly, so you would need to try `sid` for Debian.

**Reply by fwilhe2** [reply]
Does that refer to the host or the system built by mkosi? Changing testing to sid should be fine.

**Reply by daandemeyer** [reply]
@fwilhe2 The system built by mkosi, so you can change that to sid and it should resize the filesystem as well

**Reply by fwilhe2** [reply]
Yes, can confirm it works with sid.

Will look into the `directory` option, too.

Thanks for the support :100:

**Reply by ikitayama** [reply]
> Put the following in `mkosi.extra/usr/lib/repart.d/10-root.conf`:
> 
> ```
> [Partition]
> Type=root
> ```
> 
> That should be sufficient (you might need a somewhat recent systemd as I remember a recent bugfix for this).
> 
> Although for dev vms using mkosi I would skip the `disk` output completely and just use the `directory` output instead which will use virtiofs so there will be no need to select a size in the first place.

I keep hitting this Issue so @DaanDeMeyer , with the latest mkosi, the user just `$ mkdir mkosi.repart` and put a .conf file there and build together with the "main" mkosi.conf file? It took a while for me to appreciate how it is implemented thus this comment.
